src/service/referral.service.ts

```diff
@@ -55,6 +55,23 @@ class ReferralService {
 
     async createReferral(createReferralDto: CreateReferralDto): Promise<Referral> {
         const { name, email, experience, address, roleId, phone, openingId, referredById, resume } = createReferralDto;
+
+        
+        const referrals = await this.referralRepository.findReferralsByEmail(email)
+        if (referrals){
+            const referralWithSameRole = referrals.find((referral)=>referral.role.id == roleId)
+            if (referralWithSameRole){
+                let currentDate = new Date() as any
```

Comment:
Why did you cast to any? 

const currentDate = new Date();

Won't this work?